src/python/pants/backend/python/lint/pylint/rules.py

```diff
@@ -267,20 +271,23 @@ async def pylint_lint(
 
     # We batch targets by their interpreter constraints to ensure, for example, that all Python 2
     # targets run together and all Python 3 targets run together.
+    # Note that Pylint uses the AST of the interpreter that runs it. So, we include any plugin
+    # targets in this interpreter constraints calculation.
     interpreter_constraints_to_target_setup = defaultdict(set)
     for field_set, tgt, dependencies in zip(
         request.field_sets, linted_targets, per_target_dependencies
     ):
         target_setup = PylintTargetSetup(field_set, Targets([tgt, *dependencies]))
-        interpreter_constraints = (
-            PexInterpreterConstraints.create_from_compatibility_fields(
-                (
-                    *(tgt.get(PythonInterpreterCompatibility) for tgt in [tgt, *dependencies]),
-                    *plugin_targets_compatibility_fields,
+        interpreter_constraints = PexInterpreterConstraints.create_from_compatibility_fields(
+            (
+                *(
+                    tgt[PythonInterpreterCompatibility]
+                    for tgt in [tgt, *dependencies]
+                    if tgt.has_field(PythonInterpreterCompatibility)
```

Comment:
Is this change part of the fix that is effected in this PR? Or was this always correct and we just didn't do it before?

Reply:
Oh, sorry I forgot to explain. I think this is a legit bug fix. Before, if you had non-Python targets included in this iterator, we would use the default `--python-setup-interpreter-constraints`. For example, we'd do that if you had a `files()` target!

Reply:
This `if tgt.has_field()` check *feels* pretty pythonic to me too.